process/smartContract/builtInFunctions/factory.go

```diff
@@ -58,6 +62,12 @@ func CreateBuiltInFunctionsFactory(args ArgsCreateBuiltInFunctionContainer) (vmc
 	if check.IfNil(args.GuardedAccountHandler) {
 		return nil, process.ErrNilGuardedAccountHandler
 	}
+	if check.IfNil(args.PubKeyConverter) {
+		return nil, core.ErrNilPubkeyConverter
+	}
+	if len(args.WhiteListedCrossChainAddresses) == 0 {
```

Comment:
Maybe also check all addresses are valid (not nil, maybe allow only SC addresses)

Reply:
1. Validity is already being checked below, when decoding addresses
2. We need one address for each shard, not only one